editor/src/components/canvas/ui-jsx.test-utils.tsx

```diff
@@ -352,32 +353,103 @@ export async function renderTestEditorWithModel(
 
     // run dom walker
 
-    const domWalkerResult = runDomWalker({
-      domWalkerMutableState: domWalkerMutableState,
-      selectedViews: workingEditorState.patchedEditor.selectedViews,
-      elementsToFocusOn: workingEditorState.patchedEditor.canvas.elementsToRerender,
-      scale: workingEditorState.patchedEditor.canvas.scale,
-      additionalElementsToUpdate:
-        workingEditorState.patchedEditor.canvas.domWalkerAdditionalElementsToUpdate,
-      rootMetadataInStateRef: {
-        current: workingEditorState.patchedEditor.domMetadata,
-      },
-    })
+    {
+      const domWalkerResult = runDomWalker({
+        domWalkerMutableState: domWalkerMutableState,
+        selectedViews: workingEditorState.patchedEditor.selectedViews,
+        elementsToFocusOn: workingEditorState.patchedEditor.canvas.elementsToRerender,
+        scale: workingEditorState.patchedEditor.canvas.scale,
+        additionalElementsToUpdate:
+          workingEditorState.patchedEditor.canvas.domWalkerAdditionalElementsToUpdate,
+        rootMetadataInStateRef: {
+          current: workingEditorState.patchedEditor.domMetadata,
+        },
+      })
 
-    if (domWalkerResult != null) {
-      const saveDomReportAction = saveDOMReport(
-        domWalkerResult.metadata,
-        domWalkerResult.cachedPaths,
-        domWalkerResult.invalidatedPaths,
-      )
-      recordedActions.push(saveDomReportAction)
-      const editorWithNewMetadata = editorDispatch(
-        asyncTestDispatch,
-        [saveDomReportAction],
-        workingEditorState,
-        spyCollector,
-      )
-      workingEditorState = editorWithNewMetadata
+      if (domWalkerResult != null) {
+        const saveDomReportAction = saveDOMReport(
+          domWalkerResult.metadata,
+          domWalkerResult.cachedPaths,
+          domWalkerResult.invalidatedPaths,
+        )
+        recordedActions.push(saveDomReportAction)
+        const editorWithNewMetadata = editorDispatch(
+          asyncTestDispatch,
+          [saveDomReportAction],
+          workingEditorState,
+          spyCollector,
+        )
+        workingEditorState = editorWithNewMetadata
+      }
+    }
+
+    // true-up groups if needed
+    if (workingEditorState.unpatchedEditor.trueUpGroupsForElementAfterDomWalkerRuns.length > 0) {
+      ;(() => {
```

Comment:
Is this in a lambda just so that `return` can be used at line 407?

Reply:
yep. I tried to keep this code as similar to the one in editor.tsx as possible. I think eventually we should merge the two functions and the similar shape should help us match what's there vs what's here